src/Bicep.Core/Analyzers/Linter/LinterAnalyzer.cs

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Interfaces;
+using Bicep.Core.Semantics;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Globalization;
+using System.Linq;
+using System.Reflection;
+
+namespace Bicep.Core.Analyzers.Linter
+{
+    internal class LinterAnalyzer : IBicepAnalyzer
+    {
+        public static string AnalyzerName => "Bicep Internal Linter";
+        public static readonly bool IsCliInvoked;
```

Comment:
>IsCliInvoked [](http://example.com/codeflow?start=36&length=12)

In the future we will publish Bicep Core as a nuget. Would running analyzers inside that count as CLI or editing? Or would we need some 3rd option here?

We could make it a string property called, say, `hostId` or just `host`. The CLI and language servers are the two hosts we provide to everyone and each one could pass some known value for the `host` property. By default the rules would run everywhere unless someone opts it out form a host. If someone references our future nuget, they would set the `host` to whatever string they like and do the same in their linter configuration files.